extensions/sns/src/main.rs

```diff
@@ -68,49 +76,52 @@ enum SubCommand {
 }
 
 /// Executes `dfx sns` and its subcommands.
-fn main() -> anyhow::Result<()> {
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
     let opts = SnsOpts::parse();
 
-    match opts.subcmd {
-        SubCommand::DeployTestflight(args) => {
-            deploy_testflight(args);
-            Ok(())
-        }
-        SubCommand::InitConfigFile(args) => {
-            init_config_file::exec(args);
-            Ok(())
-        }
-        SubCommand::PrepareCanisters(args) => {
-            prepare_canisters::exec(args);
-            Ok(())
-        }
-        SubCommand::Propose(args) => {
-            propose::exec(args);
-            Ok(())
-        }
+    let subcommand: SnsLibSubCommand = match opts.subcmd {
+        SubCommand::DeployTestflight(args) => SnsLibSubCommand::DeployTestflight(args),
+        SubCommand::InitConfigFile(args) => SnsLibSubCommand::InitConfigFile(args),
+        SubCommand::PrepareCanisters(args) => SnsLibSubCommand::PrepareCanisters(args),
+        SubCommand::Propose(args) => SnsLibSubCommand::Propose(args),
         SubCommand::NeuronIdToCandidSubaccount(args) => {
-            neuron_id_to_candid_subaccount::exec(args);
-            Ok(())
+            SnsLibSubCommand::NeuronIdToCandidSubaccount(args)
         }
+        SubCommand::AddSnsWasmForTests(args) => SnsLibSubCommand::AddSnsWasmForTests(args),
+        SubCommand::List(args) => SnsLibSubCommand::List(args),
```

Comment:
IIUC, this converts from Foo to FooPrime. This _seems_ like an extra+unnecessary hoop. Why does FooPrime exist? Can you at least add a comment that says most of the branches in this match just convert from Foo to FooPrime?

Reply:
Added a comment explaining the value of doing this!